joomla/joomla-cms
# [4.1] Automating the rmrf on the startup of the (npm) build tools

Pull Request for Issue  #35198 .


### Summary of Changes


With #35198 we introduced a registry for all the folders of the media folder. This PR just automates the creation/update of that registry.


### Testing Instructions

This PR needs `npm`

- modify the `build/build.js` so that line 157 to line 171 are muted eg: 

```js
// Prepare the repo for dev work
if (Program.prepare) {
    const bench = new Timer('Build');
    allowedVersion();
    recreateMediaFolder(options)
    // .then(() => cleanVendors())
    // .then(() => localisePackages(options))
    // .then(() => Promise.all(
    //   [
    //     patchPackages(options),
    //     minifyVendor(),
    //     createErrorPages(options),
    //     stylesheets(options, Program.args[0]),
    //     scripts(options, Program.args[0]),
    //     bootstrapJs(),
    //     mediaManager(true)
    //   ]))
    // .then(() => bench.stop('Build'))
    // .then(() => { process.exit(0); })
    .catch((err) => {
      // eslint-disable-next-line no-console
      console.error(err);
      process.exit(-1);
    });
}
```
and also mute line 73 of `build/build-modules-js/init/recreate-media.es6.js` `// await copy(join(RootPath, 'build/media_source'), join(RootPath, 'media'), { filter: filterFunc });`

- Run `node ./build/build.js --prepare`
- Check that all the existing folders in the `media` are empty and any non-core Joomla extensions are unaffected

### Actual result BEFORE applying this Pull Request



### Expected result AFTER applying this Pull Request



### Documentation Changes Required

@richard67 this is the missing part you were asking at #35198

@wilsonge @HLeithner ~~this one actually needs to run with a GitHub action on every merge, let me know if you want me to add the action here or is something for you...~~
